### PR TITLE
Use exit code from child process

### LINKED ISSUE #44: Conduit swallows exit codes
I'm using conduit to supply `VCAP_SERVICES` to a bash script which does some processing. If the bash script fails conduit is swallowing the exit code.

Conduit should return the exit coder from my subcommand so that I can perform additional actions if required.

Here's a simple examples:

`exit.sh`
```bash
#!/bin/bash
exit 1
```

Normal operation:
```
$ ./exit.sh
$ echo $?
1
```
Wrapped by conduit:
```
$ cf7 conduit db -- ./exit.sh
OK Connecting client
...
OK Waiting for conduit app to become available
OK Fetching service infomation
OK Binding db
OK Fetching environment
OK Starting port forwarding
OK Waiting for port forwarding
OK Preparing command: ./exit.sh
$ echo $?
0
```

### PR DESCRIPTION
**Pull request by tlwr**
What
----

Fixes #44

Instead of using an empty channel and never reporting the exit code, actually pass the exit code back.

How to review
----

Code review

If this is good to merge then I will do the release dance

Who can review
----

Not @tlwr

### CODE REVIEW

File: conduit.go [hunk 1]
```diff
@@ -116,7 +116,8 @@ var ConnectService = &cobra.Command{
 		select {
 		case <-shutdown:
 			return nil
-		case <-finish:
+		case exitCode := <-finish:
+			os.Exit(exitCode)
```
Comment (chrisfarms) [inline conduit.go:120]: Is there a gotcha here from exiting early if someone comes along and wants to use the  `PostRunXXXX` family of things? ... not a big deal given the scope of this app, but if it's not _that_ important that the exact error code returned to caller, maybe wrapping/returning an `error` as "cmd xxx exited with status yyy" would give more infomation while not changing cobra's hooks stuff

Comment (tlwr) [reply conduit.go:120]: I think returning the same exit code is good user experience, but you're correct that this does interfere with the cobra pre/post

I could:

- Return an error type with an additional exit code
- Check the type in`plugin.go` around line 50
- Exit 0 or `err.ExitCode` depending

How does that suit?

Comment (chrisfarms) [reply conduit.go:120]: yeah pulling exit code out of an error type sounds more "correct" that exiting early :+1:

**Comment by tlwr** [comment]
Yeah, because we're using Cobra without any external state, our return value is always `error` \o/

**Pull request merged**